[stable/airflow] Configurable web and scheduler pod annotaitons

somcsel opened the pull request:
<!--
Thank you for contributing to helm/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/helm/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/helm/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

#### What this PR does / why we need it:

Allows annotations to be set on the Web and Scheduler Pods
This enables, for instance, the AWS IAM Role for the Pod to be specified, if using kube2iam/kiam.

#### Which issue this PR fixes
N/A

#### Special notes for your reviewer:

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md

Commit:
[stable/airflow] Fix whitespace in configuration table

Signed-off-by: Steve Larkin <steve.larkin@sony.com>

maver1ck commented on stable/airflow/README.md:
```diff
@@ -274,7 +274,8 @@ The following table lists the configurable parameters of the Airflow chart and t
 | `airflow.podDisruptionBudget`            | control pod disruption budget                           | `{'maxUnavailable': 1}`   |
-| `airflow.secretsMapping`		           | override any environment variable with a secret	     | 				             |
+| `airflow.secretsMapping`                 | override any environment variable with a secret         |                           |
+| `airflow.pod.annotations`                | annotations for the web and scheduler pods              | `{}`                      |
```
By standard we're using podAnnotations config

gsemet replied:
wouldn't it make sense to have 2 types of annotation (one for workers, one for scheduler)?

somcsel replied:
I agree: it makes sense to have different annotations for the workers - especially if adding IAM Role annotations (for kiam/kube2iam) is a common use-case.  I wouldn't want the web Pods to have the same IAM permissions as the Workers.

There is a separate set of annotations for the workers already - at line https://github.com/helm/charts/pull/11520/files#diff-57999b743b4027d904b342d1f6fbe507R283.
(Apologies if it doesn't follow the conventions - I added this in an earlier change)

How about we change to adopt the convention by using `.Values.podAnnotations` for the Web and Scheduler Pods and keep the `workers.pod.annotations` just for the workers?

somcsel replied:
(We could also rename `workers.pod.annotations` to match the standard, but that would be a compatibility break).

gsemet replied:
I do not have a strong opinion on it.
- it look like good practice says "podAnnotation" and that's ok for me
- we already have worker.pod.annotation so we would need to keep consistency if we were to add annotation for scheduler and web".

So I propose:
- airflow.podAnnotation: annotation for ALL pods (workers, scheduler, web)
- keep "worker.pod.annotation" 
- add "scheduler.pod.annotation" for the scheduler and "web.pod.annotation" for the web server

somcsel replied:
Thanks for your reviews.  I have implemented the first 2 of @gsemet's proposed changes - please take another look!

Regarding the 3rd bullet above (specific annotations for the scheduler and web Pods): do we see a use-case for this?  I'd expect it to be more common to have a specific annotaiton on the workers.  If you would like I can add this, otherwise we can wait until the requirement comes up.

maver1ck commented on stable/airflow/templates/deployments-scheduler.yaml:
```diff
@@ -27,6 +27,9 @@ spec:
       annotations:
         checksum/config: {{ include (print $.Template.BasePath "/configmap-airflow.yaml") . | sha256sum }}
         configmap.fabric8.io/update-on-change: "{{ template "airflow.fullname" . }}-env"
+        {{- range $key, $value := .Values.airflow.pod.annotations }}
```
```
{{- if .Values.podAnnotations }}
{{ toYaml .Values.podAnnotations | indent 8 }}
{{- end }}
```

maver1ck commented on stable/airflow/values.yaml:
```diff
@@ -128,6 +128,13 @@ airflow:
   connections: {}
+  ##
+  ## Annotations for the Web and Scheduler Pods
+  pod:
```
Please look at Kibana Chart as a source of inspiration
https://github.com/helm/charts/blob/master/stable/kibana/templates/deployment.yaml

Commit:
[stable/airflow] Step Chart version and describe upgrade path

Signed-off-by: Steve Larkin <steve.larkin@sony.com>

Commit:
[stable/airflow] workers.pod.annotations => workers.podAnnotations

This follows the standard used elsewhere, such as in the Kibana chart.
Note that this introduces a break in backwards compatibility.

Signed-off-by: Steve Larkin <steve.larkin@sony.com>

Commit:
[stable/airflow] Add annotations for Web/Scheduler/Worker Pods

Signed-off-by: Steve Larkin <steve.larkin@sony.com>

Commit:
[stable/airflow] Fix typo in README

Signed-off-by: Steve Larkin <steve.larkin@sony.com>

gsemet commented on stable/airflow/Chart.yaml:
```diff
@@ -1,6 +1,6 @@
 description: Airflow is a platform to programmatically author, schedule and monitor workflows
 name: airflow
-version: 0.17.4
+version: 0.17.5
```
since we break worker.pod.mapping, can we bump to 0.18.0?

somcsel replied:
Yes, according to semver I should have stepped this to 0.18.0 anyway, as this change adds a new feature.

gsemet commented on stable/airflow/templates/statefulsets-workers.yaml:
```diff
@@ -31,9 +31,15 @@ spec:
+{{- if .Values.workers.podAnnotations }}
+{{ toYaml .Values.workers.podAnnotations | indent 8 }}
+{{- end }}
+{{- if .Values.workers.pod.annotations }}
```
let's break it

somcsel replied:
Ok

gsemet reviewed (commented):
ok with the proposed solution. I propose to break the api of workers.pod.annotation once and for all.
